### PR TITLE
MULE-18264: Jar file remains open after reading resource

### PR DESCRIPTION
**Pull request by szaffarano**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/api/util/IOUtils.java [hunk 1]
```diff
@@ -95,7 +97,14 @@ public static InputStream getResourceAsStream(final String resourceName, final C
     if (url == null) {
       return null;
     } else {
-      return url.openStream();
+      URLConnection urlConnection = url.openConnection();
+      // It's necessary to disable connection caching when working with jar files
+      // in order to avoid file leaks in windows environments
+      // see for example SE-15297
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/api/util/IOUtils.java:103]: do not reference SE issues in CE code

**Comment by gsfernandes** [comment]
Check sonar issues

**Comment by szaffarano** [comment]
--validate

**Comment by szaffarano** [comment]
--validate

**Pull request merged**